File: src/main/webui/src/model/common/Customizer.ts
import CustomizerCategory from './CustomizerCategory';
import Modifiers from './Modifiers';

export default class Customizer {
    name:string;
    label:string;
    modifiers: Modifiers;
    category: CustomizerCategory;
    constructor(name:string, label:string, modifiers:Modifiers, category:CustomizerCategory) {
        this.name = name;
        this.label = label;
        this.modifiers = modifiers;
        this.category = category;
    }
}

File: src/main/java/io/tbbc/cf/customizer/Customizer.java
package io.tbbc.cf.customizer;

import io.tbbc.cf.modifier.Modifiers;
import io.tbbc.cf.research.Research;

public record Customizer(String name, String label, CustomizerCategoryName categoryName, Modifiers modifiers,
                         Research requiredResearch) {
}
